eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -554,6 +554,11 @@ public function deleteVersion( $contentId, $versionNo )
         $this->contentGateway->deleteRelations( $contentId, $versionNo );
         $this->contentGateway->deleteVersions( $contentId, $versionNo );
         $this->contentGateway->deleteNames( $contentId, $versionNo );
+
+        if ( $this->contentGateway->getLastVersionNumber( $contentId ) == 0 )
```

Comment:
@crevillo 
I would agree, but that method is used only for creating the next version, where the returned value is incremented by one.

@joaoinacio 
Here I would rather go with `listVersionNumbers`. You can then simply check if the returned array is empty. Updating the tests would also be good :)